src/category_theory/category.lean

```diff
@@ -84,6 +84,28 @@ section
 variables {C : Sort u} [𝒞 : category.{v} C] {X Y Z : C}
 include 𝒞
 
+lemma eq_of_comp_left_eq {f g : X ⟶ Y} (h : ∀ {Z : C} (h : Y ⟶ Z), f ≫ h = g ≫ h) : f = g :=
+by { convert h (𝟙 Y), tidy }
+lemma eq_of_comp_right_eq {f g : Y ⟶ Z} (h : ∀ {X : C} (h : X ⟶ Y), h ≫ f = h ≫ g) : f = g :=
+by { convert h (𝟙 Y), tidy }
+
+lemma eq_of_comp_left_eq' (f g : X ⟶ Y) (w : (λ {Z : C} (h : Y ⟶ Z), f ≫ h) = (λ {Z : C} (h : Y ⟶ Z), g ≫ h)) : f = g :=
+eq_of_comp_left_eq (λ Z h, begin
```

Comment:
I think it might be cleaner to remove the `begin ... end`. After all, `have` works just as well in term-mode.

Reply:
Don't I have to state the type explicitly to use a term-mode `have`? That would be gross.